pandas/_typing.py

```diff
@@ -22,3 +22,6 @@
                              Timedelta)
 Dtype = Union[str, np.dtype, ExtensionDtype]
 FilePathOrBuffer = Union[str, Path, IO[AnyStr]]
+
+# Type for index and columns of DataFrame
+Axes = Iterable[Union[ABCIndexClass, Iterable[str]]]
```

Comment:
@jreback is it alright? Couldn't figure out what `IUnion` is, so guessing it was a typo from you.
Also let me know if you want me write something else in the comment above.

Reply:
`Axes = Union[int, str]` - should only accept 0, 1, 'index' or 'columns'

Reply:
Ignore previous comment - late at night didn't read closely...